### PR TITLE
Changes for publishing ssz to crates.io

### LINKED ISSUE #422: Publish SSZ crate on crates.io
## Description

Our SSZ crate should be published on crates.io so it can be used in other projects (e.g., Scout)

## Steps to Resolve

- Solve the conflict between our crate and the `ssz` crate Parity already published on crates.io
- Publish it.

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

Resolves #422

## Proposed Changes

Adds licenses, comments to `ssz` and `ssz_derive`.

## Additional Info

Crates are now published on:

- https://crates.io/crates/eth2_ssz
- https://crates.io/crates/eth2_ssz_derive

### CODE REVIEW

File: eth2/utils/ssz_derive/tests/tests.rs [hunk 1]
```diff
@@ -1,22 +0,0 @@
-use ssz::Encode;
```
Comment (michaelsproul) [inline eth2/utils/ssz_derive/tests/tests.rs]: Why not keep this test?

Comment (paulhauner) [reply eth2/utils/ssz_derive/tests/tests.rs]: It creates a cyclic dependency, which means we can't upload it crates.io.

See: https://github.com/rust-lang/cargo/issues/4242

(Note the issue number!)

Comment (paulhauner) [reply eth2/utils/ssz_derive/tests/tests.rs]: Let me know if you're OK with this :)

Comment (michaelsproul) [reply eth2/utils/ssz_derive/tests/tests.rs]: Oh yeah, I'm fine with this. I just got confused about how `ssz_derive` doesn't already depend on `ssz`, but I get it now (the derive macro just knows about syntax).

**Pull request merged**